packages/headless/src/features/static-filter-set/static-filter-set-actions.ts

```diff
@@ -0,0 +1,63 @@
+import {createAction} from '@reduxjs/toolkit';
+import {validatePayload} from '../../utils/validate-payload';
+import {
+  staticFilterIdSchema,
+  staticFilterValueSchema,
+  staticFilterValuesSchema,
+} from './static-filter-set-schema';
+import {StaticFilterValue} from './static-filter-set-state';
+
+interface RegisterStaticFilterActionCreatorPayload {
+  /**
+   * A unique identifier for the static filter.
+   */
+  id: string;
+
+  /**
+   * The values of the static filter.
+   */
+  values: StaticFilterValue[];
+}
+
+export const registerStaticFilter = createAction(
+  'staticFilter/register',
+  (payload: RegisterStaticFilterActionCreatorPayload) => {
+    const schema = {
+      id: staticFilterIdSchema,
+      values: staticFilterValuesSchema,
+    };
+
+    return validatePayload(payload, schema);
+  }
+);
+
+interface ToggleSelectStaticFilterValueActionCreatorPayload {
+  /**
+   * The unique identifier for the static filter.
+   */
+  id: string;
+
+  /**
+   * The target static filter value.
+   */
+  value: StaticFilterValue;
+}
+
+export const toggleSelectStaticFilterValue = createAction(
+  'staticFilter/toggleSelect',
+  (payload: ToggleSelectStaticFilterValueActionCreatorPayload) => {
+    const schema = {
+      id: staticFilterIdSchema,
+      value: staticFilterValueSchema,
+    };
+
+    return validatePayload(payload, schema);
+  }
+);
+
+export const deselectAllStaticFilterValues = createAction(
```

Comment:
Should there be a `reallyReallyDeselectAllStaticFilters`.

Similar to how we have a `deselectAllFacets` ? 
You will probably need to add it for breadcrumb support anyway.

Reply:
I think there could be 3 levels to this:
* deselectAllStaticFilterValues(filter) // Includes all the values of one static filter
* deselectAllStaticFilters() // Includes all static filters
* clearAllFilters() // Includes facets, static filters, etc